clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -2804,8 +2804,8 @@ public void testListConsumerGroupsWithStates() throws Exception {
 
```

Comment:
It would be great if we could add more unit tests here:
* Could we test the cases where the version of the request supported does not support the type filter? It should throw an unsupported exception: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/requests/ListGroupsRequest.java#L53.
* We need to test the basic plumbing.
* We need to test the different variations of `ListConsumerGroupsOptions`.

Reply:
1) Added version check
3) Tested variation one withTypes and one without, should I also add them to the states methods in this PR?

> We need to test the basic plumbing. can you please elaborate?